modules/core/src/core/animation-loop.js

```diff
@@ -248,6 +258,11 @@ export default class AnimationLoop {
         return;
       }
       this.redraw();
+      if (this._resolveNextFrame) {
```

Comment:
This should happen in the `redraw` method, before `_endTimers`. I want to make sure those timers capture everything happening in the frame, including resolving this promise.